[UXE-7251] method route validator guide (#1666)
aziontech/docs: src/content/docs/pt-br/pages/guias/marketplace/integrations/method-route-validator.mdx

---
title: Method and Route Validator
description: >-
  Esta função pode ser usada para proteger uma aplicação bloqueando requisições com base na URI e método utilizados.
meta_tags: API, website, segurança, URI, método, rota, validador
namespace: docs_use_case_method_and_route_validator
permalink: /documentacao/produtos/guias/method-and-route-validator/
---

import Tag from 'primevue/tag';

<Tag severity="info" client:only="vue">
Preview
</Tag>

> A integração **Method and Route Validator** pode ser usada para proteger sua aplicação bloqueando requisições com base na URI e método utilizados. Ela funciona validando os JSON Args e verificando se a requisição coincide com os padrões definidos em qualquer um dos valores.

Reviewer: ```suggestion
A integração **Method and Route Validator** pode ser usada para proteger sua aplicação bloqueando requisições com base na URI e método utilizados. Ela funciona validando os argumentos e verificando se a requisição coincide com os padrões definidos em qualquer um dos valores.
```

Reviewer: em ingles esta so arguments

Com essa integração, você pode definir uma lista de rotas e qual método cada rota espera. Cada vez que a função é executada, ela seguirá as seguintes etapas:

> 1. Validará os argumentos passados via Args JSON.

Author: ```suggestion
1. Validará os argumentos passados para a function.
```

2. Verificará se a requisição coincide com os padrões definidos para as rotas.

>   - Se houver uma coincidência, a função realizará uma ação de bloqueio definida pelo usuário.

Reviewer: ```suggestion
  - Se uma correspondência for encontrada, a função executará uma ação de bloqueio especificada pelo usuário.
```

>   - Se não houver uma coincidência, a função pode ser configurada para continuar a requisição ou bloqueá-la.

Reviewer 2: Para "coincidência"a Lu sugeriu acima "correspondência ". Se optar pela troca, padronizar no texto todo.
```suggestion
  - Se não houver uma coincidência, a função pode ser configurada para continuar a requisição ou bloqueá-la.
```

>   - Se não houver uma coincidência, a função pode ser configurada para continuar a requisição ou bloqueá-la.

Author: ```suggestion
  - Se não for encontrada nenhuma correspondência, a função pode ser configurada para continuar a requisição ou bloqueá-la.
```

> 3. Se os Args JSON forem inválidos, ela escreverá uma mensagem de log e continuará a requisição.

Reviewer: ```suggestion
3. Se os argumentos forem inválidos, ela escreverá uma mensagem de log e continuará a requisição.
```

:::tip
Você pode procurar qualquer integração navegando pelos cards ou digitando uma palavra-chave na barra de busca.
:::

---

## Configure a integração

Depois de instalar a integração **Method and Route Validator**, você precisará concluir as etapas explicadas abaixo para configurá-la.

### Configure uma regra no Edge Firewall

Siga estes passos:

1. No canto superior esquerdo, abra o **Products menu** e selecione **Edge Firewall** na seção **Secure**.
2. Clique no botão **+ Edge Firewall**.
3. Dê um nome fácil de lembrar ao seu edge firewall.
4. Ative a opção **Edge Functions** na seção **Edge Firewall Modules**.
  - Esta ação dá acesso a edge functions no edge firewall.
5. Clique no botão **Save**.

Pronto. Agora você instanciou o edge firewall para sua função e tem acesso a edge functions em seu edge firewall.

:::caution[Atenção]
Se um produto ou módulo estiver ativado, pode gerar custos relacionados ao uso. Consulte a [página de preços](/pt-br/documentacao/produtos/precos/) para obter mais informações.
:::

### Configure uma função no Edge Firewall

Enquanto ainda estiver na página do **Edge Firewall**:

> 1. Selecione a aba **Functions Instances** e siga estas etapas:

Reviewer 2: Em EN está Select the **Functions Instances** tab.
```suggestion
1. Selecione a aba **Functions Instances**.
```

2. Clique no botão **+ Function Instance**.
3. Dê um nome fácil de lembrar à sua instância.
4. No menu suspenso, selecione a função **Method and Route Validator**.
5. Na caixa **Arguments**, passe suas variáveis, como no exemplo:

```json
{
  "restricted_mode": false,
  "action": "deny",
  "routes": [
    {
      "match_type": "equals",
      "path": "/my/route/",
      "methods": ["GET", "POST"]
    },
    {
      "match_type": "contains",
      "path": "/something/",
      "methods": ["POST"]
    },
    {
      "match_type": "regex",
      "path": "^.test.*",
      "methods": ["GET", "POST", "PATCH", "DELETE"]
    }
  ]
}
```

Onde:

| Propriedade | Tipo | Obrigatório | Descrição |
| --- | --- | --- | --- |
| `restricted_mode` | Boolean | Não | Indica se a função deve operar no modo restrito. Valor padrão: `false`. |

> | `action` | String | Sim | Define qual ação a ser usada quando a função identifica a requisição como inválida. |

Reviewer 2: Simplifiquei "a ser usada"
```suggestion
| `action` | String | Sim | Define qual ação usar quando a função identifica a requisição como inválida. |
```

| `routes` | Array | Sim | Contém todas as URIs que a aplicação protegida espera lidar. |
| `routes.match_type` | String | Sim | String que especifica o tipo de match a ser realizado no caminho. |

> | `routes.path` | String | Sim | Define o argumento a ser usado para validar a URI da requisição. |

Reviewer 2: Define qual argumento usar?
```suggestion
| `routes.path` | String | Sim | Define o argumento a ser usado para validar a URI da requisição. |
```
